pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -27,21 +27,13 @@ import (
 	"github.com/grafana/grafana/pkg/services/query"
 	"github.com/grafana/grafana/pkg/setting"
 	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/clients"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/kinds/dataquery"
 	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
 )
 
 type DataQueryJson struct {
-	QueryType       string `json:"type,omitempty"`
-	QueryMode       string
-	PrefixMatching  bool
-	Region          string
-	Namespace       string
-	MetricName      string
-	Dimensions      map[string]interface{}
-	Statistic       *string
-	Period          string
-	ActionPrefix    string
-	AlarmNamePrefix string
+	dataquery.CloudWatchAnnotationQuery
+	Type string `json:"type,omitempty"`
```

Comment:
Note: this was previously

```go
QueryType string `json:"type,omitempty"`
```

`QueryType` already comes from `common.DataQuery` https://github.com/grafana/grafana/blob/b6ddc5b35b29eee665a11c1b77568dd715343ff6/pkg/tsdb/cloudwatch/kinds/dataquery/types_dataquery_gen.go#L255, at runtime we are setting the `type` on the query for annotation queries here https://github.com/grafana/grafana/blob/b6ddc5b35b29eee665a11c1b77568dd715343ff6/public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchAnnotationQueryRunner.ts#L34.

Reply:
Am I following correctly that what you're seeing is that pretty much everything is set as an annotationQuery? 🤔 

What is `DataQueryJson` anyway? A metrics query (so maybe it should be dataquery.CloudWatchMetricsQuery? 


Or is `DataQueryJson` a type which contains some common fields? I'm seeing the following in the generated types. Could this be the one we should be using here instead? (If yes, what should we do to add the appropriate types there?)

```
type CloudWatchDataQuery = map[string]interface{}
```

Reply:
only annotation queries are set as `annotationQuery`. the other query runners set it to the appropriate type.

---

`DataQueryJson` is the shape of the query for annotation queries. there's an existing issue around refactoring this type #52001.

from this bit of the code, we only really need to read read the query type to see what type of query we need to run, so something like this could be used when we work on #52001

```go
type CloudWatchDataQuery struct {
  Type string
}
```